### PR TITLE
feat: allow builder boost factor to be configured via proposer config file

### LINKED ISSUE #6304: Support builder boost factor in proposer settings file
### Problem description

We are not supporting `boostFactor` in proposer settings file but instead only on the keymanager API
https://github.com/ChainSafe/lodestar/blob/e51f495887531d676db68955b0b0bce9b2b84f84/packages/cli/src/util/proposerConfig.ts#L14-L24

### Solution description

Add `builder.boostFactor` to proposer settings file to allow per validator configuration

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by g11tech**
allows per validator configuration of boost factor for builder via static proposer config file

Closes https://github.com/ChainSafe/lodestar/issues/6304

### CODE REVIEW

File: packages/cli/src/util/proposerConfig.ts
Comment (ensi321) [inline packages/cli/src/util/proposerConfig.ts:84]: Funny how `boost_factor` has type `bigint | undefined` yet we check for a string here. Looks confusing along with string checks on other config fields. Anyway it looks fine for the purpose of this PR.

**Review by ensi321 — APPROVED** [review]
lgtm

File: packages/cli/src/util/proposerConfig.ts
Comment (nflaig) [inline packages/cli/src/util/proposerConfig.ts]: This is the same parsing we had previously for the CLI flag but it produces unclear error if a non-integer value is passed. We should ideally reuse parsing function introduces in https://github.com/ChainSafe/lodestar/pull/6332.

https://github.com/ChainSafe/lodestar/blob/01d47b9a0901c9f03a01ef68c819f4e592cbfd8b/packages/cli/src/cmds/validator/handler.ts#L290

or something along those lines

File: packages/cli/src/util/proposerConfig.ts
Comment (g11tech) [reply packages/cli/src/util/proposerConfig.ts:84]: bigint is jsoned as a string as its not number safe

File: packages/cli/src/util/proposerConfig.ts
Comment (nflaig) [inline packages/cli/src/util/proposerConfig.ts]: Noticed we might wanna also improve validation on builder selection
https://github.com/ChainSafe/lodestar/blob/01d47b9a0901c9f03a01ef68c819f4e592cbfd8b/packages/cli/src/cmds/validator/handler.ts#L257

Might be better to define those functions here and import them in the handler instead of the other way around

**Review by nflaig — APPROVED** [review]

File: packages/cli/src/util/proposerConfig.ts
Comment (nflaig) [inline packages/cli/src/util/proposerConfig.ts:130]: Just came to my mind that we might wanna use a normal Error here now as this can be thrown from multiple places and not just strictly from the CLI (yargs). Having a stacktrace might also be helpful for users to identify the error in that case.

I can address this in another PR, wanted to look into adding `executionalways` but this is currently conflicts with https://github.com/ChainSafe/lodestar/pull/6241, so makes sense to wait on that until that is merged.

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.16.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.16.0) :tada:

**Pull request merged**
